cli/exchange/nmp.go

```diff
@@ -0,0 +1,157 @@
+		`  "properties": [                            /* ` + msgPrinter.Sprintf("(Optional) A list of policy properties that describe this policy.") + ` */`,
+		`    {`,
+		`      "name": "",`,
+		`      "value": null`,
+		`    }`,
+		`  ],`,
+		`  "constraints": [                           /* ` + msgPrinter.Sprintf("(Optional) A list of constraint expressions of the form <property name> <operator> <property value>,") + ` */`,
+		`    "myproperty == myvalue"                  /* ` + msgPrinter.Sprintf("separated by boolean operators AND (&&) or OR (||).") + `*/`,
+		`  ],`,
+		`  "patterns": [                              /* ` + msgPrinter.Sprintf("(Optional) This policy applies to nodes using one of these patterns.") + ` */`,
+		`    ""`,
+		`  ],`,
+		`  "enabled": false,                          /* ` + msgPrinter.Sprintf("Is this policy enabled or disabled.") + ` */`,
+		`  "agentUpgradePolicy": {                    /* ` + msgPrinter.Sprintf("(Optional) Assertions on how the agent should update itself.") + ` */`,
+		`    "atLeastVersion": "<version> | current", /* ` + msgPrinter.Sprintf("Specify the minimum agent version these nodes should have, default \"current\".") + ` */`,
+		`    "start": "<RFC3339 timestamp> | now",    /* ` + msgPrinter.Sprintf("When to start an upgrade, default \"now\".") + ` */`,
+		`    "duration": 0                            /* ` + msgPrinter.Sprintf("Enable agents to randomize upgrade start time within start + duration, default 0.") + ` */`,
+		`  }`,
+		`}`,
+	}
+
+	for _, s := range nmp_template {
+		fmt.Println(s)
+	}
+}
+
+func NMPAdd(org, credToUse, nmpName, jsonFilePath string, noConstraints bool) {
+	// check for ExchangeUrl early on
+	var exchUrl = cliutils.GetExchangeUrl()
+
+	cliutils.SetWhetherUsingApiKey(credToUse)
+	var nmpOrg string
+	nmpOrg, nmpName = cliutils.TrimOrg(org, nmpName)
+
+	// get message printer
+	msgPrinter := i18n.GetMessagePrinter()
+
+	// read in the new nmp from file
+	newBytes := cliconfig.ReadJsonFileWithLocalConfig(jsonFilePath)
+	var nmpFile exchangecommon.ExchangeNodeManagementPolicy
+	err := json.Unmarshal(newBytes, &nmpFile)
+	if err != nil {
+		cliutils.Fatal(cliutils.JSON_PARSING_ERROR, msgPrinter.Sprintf("failed to unmarshal json input file %s: %v", jsonFilePath, err))
+	}
+
+	// validate the format of the nmp
+	err = nmpFile.Validate()
+	if err != nil {
+		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("Incorrect node management policy format in file %s: %v", jsonFilePath, err))
+	}
+
+	// if the --no-constraints flag is not specified and the given nmp has no constraints, alert the user.
+	if (!noConstraints) && nmpFile.HasNoConstraints() {
+		cliutils.Fatal(cliutils.CLI_INPUT_ERROR, msgPrinter.Sprintf("The node management policy has no constraints which might result in the service being deployed to all nodes. Please specify --no-constraints to confirm that this is acceptable."))
```

Comment:
"result in the service being deployed to all nodes" should be changed to "result in the management policy being deployed to all nodes"